### ISSUE TITLE
stackplot() toggles the matplotlib backend

### ISSUE THREAD

**Issue by CommonClimate** [LinkedEarth/Pyleoclim_util#282]
**Describe the bug**
in a Jupyter setting, invoking `stackplot()` toggles the matplotlib backend so that inline figures no longer display. The stackplot displays normally, but none of the subsequent figures do, unless `%matplotlib inline` magic is invoked.

**To Reproduce**
e.g. PyleoTutorials L1_figures_with_multiple_panels.ipynb, though has been observed in any notebook 

**Expected behavior**
Backend is preserved. 

**Hypothesis**
the problem appears after display, so it is probably related to the resetting of the original style, as per `            mpl.rcParams.update(current_style)`. However, I was not able to figure out why this command messes with the [backend](https://matplotlib.org/stable/users/explain/backends.html). @fzhu2e can you help?

**Version**
 - Python 3.10
 - Matplotlib 3.6.2
 - Pyleoclim Development version (0.9.2b)

**Additional context**
Jupyter context

**Comment by CommonClimate** [comment]
Possible solution: use the "context" feature of Matplotlib styles within stackplot() instead of messing with the rcParams by applying changes to the global style. This would require defining our styles as proper `.mplstyle` files, which makes sense for re-use outside of Pyleoclim.

**Comment by fzhu2e** [comment]
It's indeed strange that `mpl.rcParams.update(current_style)` makes the inline mode of a Jupyter notebook environment not working, unless explicitly invoked. The idea of `current_style` is to save the current style being used, which might not be our predefined ones; could be other customized ones. Saving the current style to a file is cumbersome anyway.

After testing, I find it not quite necessary to set the style inside `stackplot()`. I did that to force it to be `journal` style and made modifications based on that. Now I find that, even with a `web` style, it doesn't look bad. So in commit 275ac9d, I just commented out those lines. So that the "current style" will not be forced to change, and there's no need to `mpl.rcParams.update(current_style)` anymore.

**Comment by CommonClimate** [comment]
Thanks Feng. Can you confirm that this addresses the Jupyter backend behavior with Matplotlib >=3.6?

Julien Emile-Geay, associate professor
University of Southern California


On Dec 15, 2022, at 23:51, Feng Zhu ***@***.***> wrote:

﻿

It's indeed strange that mpl.rcParams.update(current_style) makes the inline mode of a Jupyter notebook environment not working, unless explicitly invoked. The idea of current_style is to save the current style being used, which might not be our predefined ones; could be other customized ones. Saving the current style to a file is cumbersome anyway.

After testing, I find it not quite necessary to set the style inside stackplot(). I did that to force it to be journal style and made modifications based on that. Now I find that, even with a web style, it doesn't look bad. So in commit 275ac9d<https://urldefense.com/v3/__https://github.com/LinkedEarth/Pyleoclim_util/commit/275ac9dd366bd90204b084ab9003e7c4a6ca7a06__;!!LIr3w8kk_Xxm!to8-EgqWDVZz4G1nX1FC05ecZ5DwILEUTelX29W73SjpiIwpsX8HP-nRPsmt--Rz2vZWz7y1LIlpZFzR2f5jqlwB$>, I just commented out those lines. So that the "current style" will not be forced to change, and there's no need to mpl.rcParams.update(current_style) anymore.

**Comment by fzhu2e** [comment]
Yes, I tested under py3.10 with matplotlib 3.6.2.

**Comment by CommonClimate** [comment]
Great. I was able to confirm that everything works properly now, at least in the "journal" mode: https://github.com/LinkedEarth/PyleoTutorials/blob/main/notebooks/L0_basic_MSES_manipulation.ipynb.  Oddly enough, `stripes()` doesn't seem to do that toggle, even though I borrowed the same trick that `stackplot()` used to employ. I'll leave that for now, and remove it if it causes trouble down the line. Thanks @fzhu2e for the prompt action!
